cql3/statements/create_table_statement.cc

```diff
@@ -96,10 +97,31 @@ std::vector<column_definition> create_table_statement::get_columns()
     return column_defs;
 }
 
+template <typename CreateTable>
+future<shared_ptr<cql_transport::event::schema_change>>
+create_table_statement::create_table_with_cdc(service::storage_proxy& proxy,
+                                              schema_ptr schema,
+                                              CreateTable&& create_table) {
+    if (_if_not_exists) {
+        throw exceptions::invalid_request_exception(
+                "Can't create table with CDC support using IF NOT EXISTS");
+    }
+    return cdc::setup(proxy, schema).then([create_table = std::move(create_table), schema = std::move(schema)] () mutable{
+        return create_table().handle_exception([schema = std::move(schema)](std::exception_ptr ep) mutable {
```

Comment:
This confirms my previous suspicion. We really don't want `create_table` to throw and instead return an exceptional future, but it might throw as it currently is (from the `announce_new_column_family` call).

Reply:
Fixed.